src/manager.cpp

```diff
@@ -203,12 +203,101 @@ void Manager::SetTimerToDetectVpdCollectionStatus()
+    }
+
+    try
+    {
+        int l_bytesWrittenOnDbus = -1;
+
+        // For the given VPD path, get it's corresponding inventory object path
+        // from system config JSON
+        const std::string& l_inventoryObjPath =
+            jsonUtility::getInventoryObjPathFromJson(l_jsonObj, i_vpdPath);
+
+        if (!l_inventoryObjPath.empty())
+        {
+            types::DbusPropertyName l_propertyName;
+            types::DbusVariantType l_propertyValue;
+            types::DbusInterface l_interfaceName;
+
+            // Extract data from input
+            if (const types::IpzData* l_ipzData =
+                    std::get_if<types::IpzData>(&i_paramsToWriteData))
+            {
+                types::Record l_recordName = std::get<0>(*l_ipzData);
+                l_interfaceName = constants::ipzVpdInf + l_recordName;
+
+                l_propertyName = std::get<1>(*l_ipzData);
+
+                types::BinaryVector l_keywordData = std::get<2>(*l_ipzData);
+                l_propertyValue.emplace<types::BinaryVector>(l_keywordData);
+            }
+
+            // Update the keyword's value on D-bus inventory object path.
+            l_bytesWrittenOnDbus = writeOnDbus(l_inventoryObjPath,
+                                               l_interfaceName, l_propertyName,
+                                               l_propertyValue);
+        }
+
+        int l_bytesWrittenOnHardware = -1;
+
+        // For the given VPD path, get it's corresponding hardware path from
+        // system config JSON.
+        const std::string& l_fruPath =
+            jsonUtility::getFruPathFromJson(l_jsonObj, i_vpdPath);
+
+        if (!l_fruPath.empty())
+        {
+            // Update the keyword's value on hardware path.
+            l_bytesWrittenOnHardware = writeOnHardware(l_fruPath, l_jsonObj,
+                                                       i_paramsToWriteData);
+        }
+
+        // If update on D-bus and hardware succeeds, store the actual no. of
+        // bytes written.
+        if (l_bytesWrittenOnHardware == l_bytesWrittenOnDbus)
+        {
+            logging::logMessage("VPD write successful on " + l_fruPath);
+            logging::logMessage("VPD write successful on " +
+                                l_inventoryObjPath);
+
+            l_bytesWritten = l_bytesWrittenOnHardware;
+        }
+        else
+        {
+            // TODO:Handle the cases where either one of the writes fail.
+        }
+
+        // For the given VPD path, get it's corresponding redundant hardware
+        // path from system config JSON
+        const std::string& l_redundantFruPath =
+            jsonUtility::getRedundantEepromPathFromJson(l_jsonObj, i_vpdPath);
+
+        // Check if there is a redundant hardware path for the given VPD path.
+        if (!l_redundantFruPath.empty())
+        {
+            // Update the keyword's value on redundant hardware path
+            if (writeOnHardware(l_redundantFruPath, l_jsonObj,
```

Comment:
since its a redundant eeprom, even if write fails, it won't affect the running system. so we can log the error and return successfully

Reply:
Even if writing on Bus fails, it won't affect the running system. It will just create a mismatch.
The whole idea is to keep all the copy of data in sync, be it primary/redundant or cache. If redundant only fails implies it is no longer redundant.